zkevm-circuits/src/table.rs

```diff
@@ -2229,6 +2229,193 @@ impl<F: Field> LookupTable<F> for SigTable {
 }
 
+/// Lookup table embedded in the modexp circuit for precompile.
+#[derive(Clone, Copy, Debug)]
+pub struct ModExpTable {
+    /// Use for indicate beginning of a limbs group
+    pub q_head: Column<Fixed>,
+    /// base represented by limbs
+    pub base: Column<Advice>,
+    /// exp represented by limbs
+    pub exp: Column<Advice>,
+    /// modulus represented by limbs
+    pub modulus: Column<Advice>,
+    /// result represented by limbs
+    pub result: Column<Advice>,
+}
+
+impl ModExpTable {
+    /// Construct the modexp table.
+    pub fn construct<F: Field>(meta: &mut ConstraintSystem<F>) -> Self {
+        let ret = Self {
+            q_head: meta.fixed_column(),
+            base: meta.advice_column(),
+            exp: meta.advice_column(),
+            modulus: meta.advice_column(),
+            result: meta.advice_column(),
+        };
+        meta.enable_equality(ret.base);
+        meta.enable_equality(ret.exp);
+        meta.enable_equality(ret.modulus);
+        meta.enable_equality(ret.result);
+        ret
+    }
+
+    /// helper for devide a U256 into 3 108bit limbs
+    pub fn split_u256_108bit_limbs(word: &Word) -> [u128; 3] {
+        let bit108 = 1u128 << 108;
+        let (next, limb0) = word.div_mod(U256::from(bit108));
+        let (limb2, limb1) = next.div_mod(U256::from(bit108));
+        [limb0.as_u128(), limb1.as_u128(), limb2.as_u128()]
+    }
+
+    /// helper for obtain the modulus of a U256 in Fr
+    pub fn native_u256<F: Field>(word: &Word) -> F {
+        let minus1 = -F::one();
+        let (div, _) = word.div_mod(Word::from_little_endian(minus1.to_repr().as_ref()));
+        let div = div.checked_add(Word::from(1u64)).unwrap();
+        let mut bytes = [0u8; 64];
+        div.to_little_endian(&mut bytes[..32]);
+        F::from_bytes_wide(&bytes)
+    }
+
+    /// fill a blank 4-row region start from offset for empty lookup
+    pub fn fill_blank<F: Field>(&self, layouter: &mut impl Layouter<F>) -> Result<(), Error> {
+        layouter.assign_region(
+            || "modexp table blank region",
+            |mut region| {
+                for i in 0..4 {
+                    // fill last totally 0 row
+                    region.assign_fixed(
+                        || "modexp table blank head row",
+                        self.q_head,
+                        i,
+                        || Value::known(F::zero()),
+                    )?;
+                    for &col in [&self.base, &self.exp, &self.modulus, &self.result] {
+                        region.assign_advice(
+                            || "modexp table blank limb row",
+                            col,
+                            i,
+                            || Value::known(F::zero()),
+                        )?;
+                    }
+                }
+                Ok(())
+            },
+        )
+    }
+
+    /// Get assignments to the modexp table. Meant to be used for dev purposes.
+    pub fn dev_load<F: Field>(
```

Comment:
this method is used both (1) in super circuit, where the table is constraint by circuit (2) in evm circuit, where table is not constrained.

For some table like keccak table, (1) and (2) is different. 'dev_load' is the name only used for (2). So i suggest change `dev_load` to `load`.

Reply:
I think it should be `dev_load`. In super circuit the table is assigned [inside modexp circuit](https://github.com/scroll-tech/zkevm-circuits/blob/0cdd7c45b1d67fd7e5f1bb0528416f526ec2bc0d/zkevm-circuits/src/modexp_circuit.rs#L57)